DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+  ```
+  public static final double TRACK_WIDTH = 0.7112;
+  public static final double WHEEL_RADIUS = Units.inchesToMeters(3);
+  public static final double GEARING =  8;
+  public static final double MOI = 7.5;
+  public static final Measure<Mass> DRIVE_MASS = Kilograms.of(60.0);
+  public static final Matrix<N7, N1> STD_DEVS = VecBuilder.fill(0, 0, 0, 0, 0, 0, 0);
+  ```
+By setting up the simulation this way, we’re creating a virtual model of our robot that behaves similarly to how it would in real life.
+
+Before we start periodically updating our sim values, make sure to set the voltage to our sim drive in the `setSpeed` method: 
+```java
+driveSim.setInputs(leftVoltage, rightVoltage);
+```
+
+#### Simulation in Action: `simulationPeriodic`
+The `simulationPeriodic` method is where we update the simulation. This method is called regularly during the robot’s operation to keep the simulated sensors and drivetrain in sync with the rest of the code. We must put a `@Override` annotation above the method as it is inherited from extending to `subsystemBase`.
+```java
+@Override
+public void simulationPeriodic(){
+  driveSim.update(0.02);  // Update the drivetrain simulation every 20ms
+  leftEncoder.setPosition(driveSim.getLeftPositionMeters());
+  rightEncoder.setPosition(driveSim.getRightPositionMeters());
+  gyroSim.setAngle(-driveSim.getHeading().getDegrees());
+}
+```
+- `leftEncoder.setPosition` and `rightEncoder.setPosition`: These update the simulated encoder positions to match the simulated robot’s movement.
+- `gyroSim.setAngle`: Updates the simulated gyro angle based on the robot’s simulated heading.
+
+This method ensures that our simulated sensors provide accurate feedback as the robot "moves" in the simulation, allowing us to test and tweak our code.
+
+#### Logging: Capturing Important Data
+Now let’s talk about logging. Logging is crucial for understanding how our robot behaves over time, diagnosing issues, and improving performance. We’re going to use Monologue for logging, which gives us a structured way to record and analyze data from the robot’s systems.
+
+Here’s how we set up Monologue:
+```java
+private void configureGameBehavior() {
+  Monologue.setupMonologue(this, "/Robot", false, true);
+  addPeriodic(Monologue::updateAll, kDefaultPeriod);
+  addPeriodic(FaultLogger::update, 1);
+}
+```
+This initializes Monologue with our robot, setting up the logging system to capture data as the robot runs and makes sure Monologue logs data at regular intervals, which we’ve defined with kDefaultPeriod. By logging data regularly, we can later review how the robot performed and make informed adjustments to our code.
+
+#### Using @Log.NT for NetworkTables Logging
+We use NetworkTables to make logging specific variables or objects easier. We’re going to use the `@Log.NT` annotation as it allows us to log key metrics in real-time while the robot is running, which we can then view on dashboards like Shuffleboard or SmartDashboard.
```

Comment:
First sentence is correct but leaves out a lot. I would link to the Telemetry doc once merged